lms/services/document_url.py

```diff
@@ -32,7 +32,9 @@ def get_document_url(self, request) -> Optional[str]:
 
         return None
 
-    _ENCODED_URL = re.compile("^(?:https?|canvas|vitalsource|jstor)%3a", re.IGNORECASE)
+    # If the URL looks like "schema%3a" (schema:) it's double-encoded
+    # https://www.rfc-editor.org/rfc/rfc3986#section-3.1
+    _ENCODED_URL = re.compile("^[a-z][a-z0-9+0.-]+%3a", re.IGNORECASE)
```

Comment:
```suggestion
    _ENCODED_URL = re.compile("^[a-z][a-z0-9+.-]*%3a", re.IGNORECASE)
```

Removed a repeated "0". The change from "+" to "*" at the end is not practically important, but just for spec consistency.